tools/deploy.sh

```diff
@@ -90,30 +90,50 @@ RESTART_CONTAINER_CERTIFICATE_UPDATED=${RESTART_CONTAINER_CERTIFICATE_UPDATED:-"
 export NAMEPREFIX=${NAMEPREFIX:-"baremetal-operator"}
 
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+IRONIC_BASIC_AUTH_COMPONENT="${SCRIPTDIR}/ironic-deployment/components/basic-auth"
+TEMP_IRONIC_OVERLAY="${SCRIPTDIR}/ironic-deployment/overlays/temp"
+rm -rf "${TEMP_IRONIC_OVERLAY}"
+mkdir -p "${TEMP_IRONIC_OVERLAY}"
 
-KUSTOMIZE="tools/bin/kustomize"
+KUSTOMIZE="${SCRIPTDIR}/tools/bin/kustomize"
 make -C "$(dirname "$0")/.." "${KUSTOMIZE}"
 
+# Create a temporary overlay where we can make changes.
+pushd "${TEMP_IRONIC_OVERLAY}"
+${KUSTOMIZE} create --resources=../../../config/namespace \
+  --namespace=baremetal-operator-system --nameprefix=baremetal-operator-
+${KUSTOMIZE} edit add secret mariadb-password --from-literal=password=changeme
+
+# Basic-auth + TLS is special since TLS also means reverse proxy, which affects basic-auth.
+# Therefore we have an overlay that we use as base for this case.
+if [ "${DEPLOY_BASIC_AUTH}" == "true" ] && [ "${DEPLOY_TLS}" == "true" ]; then
+    ${KUSTOMIZE} edit add resource ../../overlays/basic-auth_tls
+else
+    ${KUSTOMIZE} edit add resource ../../base
+    if [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
+        ${KUSTOMIZE} edit add component ../../components/basic-auth
+    elif [ "${DEPLOY_TLS}" == "true" ]; then
+        ${KUSTOMIZE} edit add component ../../components/tls
+    fi
+fi
+
+if [ "${DEPLOY_KEEPALIVED}" == "true" ]; then
+    ${KUSTOMIZE} edit add component ../../components/keepalived
+fi
+
 if [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
     BMO_SCENARIO="${SCRIPTDIR}/config/basic-auth"
-    IRONIC_SCENARIO="${SCRIPTDIR}/ironic-deployment/basic-auth"
 else
     BMO_SCENARIO="${SCRIPTDIR}/config"
-    IRONIC_SCENARIO="${SCRIPTDIR}/ironic-deployment"
 fi
 
 if [ "${DEPLOY_TLS}" == "true" ]; then
     BMO_SCENARIO="${BMO_SCENARIO}/tls"
-    IRONIC_SCENARIO="${IRONIC_SCENARIO}/tls"
 elif [ "${DEPLOY_BASIC_AUTH}" == "true" ]; then
     BMO_SCENARIO="${BMO_SCENARIO}/default"
 fi
 
-if [ "${DEPLOY_KEEPALIVED}" == "true" ]; then
-    IRONIC_SCENARIO="${IRONIC_SCENARIO}/keepalived"
-else
-    IRONIC_SCENARIO="${IRONIC_SCENARIO}/default"
```

Comment:
You mean deploying with/without keepalived? No it is still there but there is no explicit "default". We just add the keepalived components if needed, otherwise we leave it.
https://github.com/metal3-io/baremetal-operator/blob/b7251a0288d581995c0f7897c96dcc822166cd82/tools/deploy.sh#L120-L122

Reply:
> You mean deploying with/without keepalived?

yeah, basically when `DEPLOY_KEEPALIVED == false`

Reply:
Yes so if `DEPLOY_KEEPALIVED` is false (or anything other than true), then we skip adding the keepalived component. This should give us the exact same as before :slightly_smiling_face: